File: backend_clickmesa/backend_clickmesa/schemas/schedule.py

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ScheduleBase(BaseModel):
    scheduled_date: datetime
    meal_type: Optional[str] = None
    portions: Optional[int] = None
    recipe_id: int


class ScheduleCreate(ScheduleBase):
    user_id: int


class ScheduleUpdate(BaseModel):
    scheduled_date: Optional[datetime] = None
    meal_type: Optional[str] = None
    portions: Optional[int] = None
    recipe_id: Optional[int] = None


class SchedulePublic(ScheduleBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True


class Message(BaseModel):
    message: str

File: frontend_clickmesa/src/types/schedule.ts

export interface ScheduleCreate {
    scheduled_date: string;
    recipe_id: number;
    user_id: number;
    meal_type:string;
    portions:number;
}

export interface SchedulePublic {
    id: number;
    scheduled_date: string;
    meal_type: string;
    portions: number;
    user_id: number;
    recipe_id: number;
    recipe_title?: string;
}

export interface ScheduledMeal {
  id: number;
  meal_type: string;
  scheduled_date: string;
  recipe_id: number;
  recipe_title?: string;
  portions: number;
}